numpy/distutils/command/config.py

```diff
@@ -101,8 +101,12 @@ def _wrap_method(self, mth, lang, args):
         return ret
 
     def _compile (self, body, headers, include_dirs, lang):
-        return self._wrap_method(old_config._compile, lang,
-                                 (body, headers, include_dirs, lang))
+        src, obj = self._wrap_method(old_config._compile, lang,
+                                     (body, headers, include_dirs, lang))
+        # _compile in unixcompiler.py sometimes creates .d dependency files.
+        # Clean them up.
+        self.temp_files.append(obj + '.d')
+        return src, obj
```

Comment:
this will remove the depency file after compilation?
that defeats the whole point of the dependency file, please revert this.

Reply:
hm nevermind it still appears to be present, what does this temp_files object actually do?

Reply:
I'd like to know also, I couldn't find it documented.

Reply:
The numpy distutils code is a set of patches into the python distutils module; whoever wrote the numpy distutils used a lot of hidden functionality inside the distutils module. I read both sets of code to figure out what to do here.

You can see `temp_files` used further below in this file to clean up the lib files, and you can also see it used in the main distutils `command/config.py` code.

I think the important fact is that this file `numpy/distutils/command/config.py` is only used in the configuration step and the config object is only created by calling `get_config_cmd`. A grep will show this is only used for temporary compilations for use in the config. That is why it is harmless to clean up the dependency files.

The `build_clib.py` file is what actually builds the numpy libraries. That class does *not* clean up the dependency files.

Reply:
the question is when does it clean them up
I assume it is intended to be used during the clean target, but we do not support that target in numpy itself
though it could be useful for third parties